source/dagger/gameplay/ping_pong/pingpong_ball.h

```diff
@@ -10,6 +13,9 @@ namespace ping_pong
     struct PingPongBall
     {
         Vector3 speed{ 0, 0, 0 };
+        
+        std::chrono::seconds changeSpeedPeriod{ 5 };
+        TimePoint lastSpeedChange{ dagger::Engine::Instance().CurrentTime() };
```

Comment:
looks fine. 

But you can implement it in more usual way:
have a float timer = 5 inside PingPongBall
and do timer -= DeltaTime each frame.
and check if it's < than a 0. 
That way you won't need to work with chrono library inside gameplay classes.

Reply:
Hmmm, [delta time](https://github.com/UbiBelETF/dagger/blob/b5ca5c344b1be7513abd06af5fdec1b82c2143bb/source/dagger/core/engine.h#L61) might help here!